manticore/core/cpu/abstractcpu.py

```diff
@@ -324,6 +385,20 @@ def __init__(self, regfile, memory):
         self._md.syntax = 0
         self.instruction = None
 
+        #####################################################
+        # Signals
+        # signal handlers must have this signature:
+        # handler(cpu, *args, **kwargs)
+        self.will_decode_instruction = Signal()
+        self.will_execute_instruction = Signal()
+        self.did_execute_instruction = Signal()
+        self.will_emulate_instruction = Signal()
+        self.did_emulate_instruction = Signal()
+        self.will_read_register = Signal()
```

Comment:
it seems like some of these are unused? can have documented use cases before adding these in?